clients/core-java/src/main/java/org/commonjava/indy/client/core/IndyClientHttp.java

```diff
@@ -607,7 +650,21 @@ public boolean validResponseCode( final int statusCode, final int[] responseCode
     public void close()
     {
         logger.debug( "Shutting down indy client HTTP manager" );
-        connectionManager.reallyShutdown();
+        if ( location != null )
+        {
+            try
+            {
+                factory.close();
```

Comment:
The 'location' (site config) is used only to determine which way to go, i.e., using Factory or httpclient. We don't make it a flag as to close or not - that would be too much for it. Actually we may add a 'closed' boolean and check it in the 'connect' method to make sure a closed client not serve any further request. @jdcasey , how do you think?